### DISCUSSION TITLE
How to browse node ID when I just know the server endpoint

### DISCUSSION THREAD

**Discussion by jjauzion** [gopcua/opcua#545] [Q&A] [answered]
I have an opcua server on let's say `opc.tcp://localhost:4840`.

Is there a way to browse for node ID when I just know the endpoint ?

The `opcua/examples/browse/` example ask for a node ID and using an empty string as the node ID, I get the following error:
`The node id refers to a node that does not exist in the server address space. StatusBadNodeIDUnknown (0x80340000)`

All the example I saw use a node ID. But I don't know the node ID of the server. How should I proceed ?

Sorry if my question is dumb but I'm new to OPCUA and don't know much about this protocol

**Comment by Adamsst** [comment, marked as answer]
OPCUA has a specification available at https://reference.opcfoundation.org/, Section 8 has information on the root object.
The root object has a NodeId of "i=84".
You can plug that NodeId to browse the entire server using the example, it may take a while as that will browser all nodes.

**Reply by jjauzion** [reply]
Awesome, thanks for the help and for the specification, that will help a lot !

**Reply by peng110653** [reply]
are you use "ns=0;1=85"?   ns=0;1=84

**Reply by peng110653** [reply]
Can you tell me which command you used?

**Reply by jjauzion** [reply]
@peng110653 I'm using with "ns=0;i=84", seems to work

**Reply by magiconair** [reply]
`i=84` is sufficient since `ns=0` is the default.
